src/views/RuleEngine/components/SQLTemplateDrawer.vue

```diff
@@ -0,0 +1,156 @@
+<template>
+  <el-drawer
+    v-model="showDrawer"
+    :title="tl('commonSQLTemplates')"
+    custom-class="SQL-template-drawer"
+  >
+    <el-collapse>
+      <el-collapse-item
+        v-for="item in templateList"
+        :key="item.title"
+        :title="item.title"
+        :name="item.title"
+      >
+        <section>
+          <div class="sub-block-hd">
+            <p class="hd-title">{{ tl('SQL') }}</p>
+            <div>
+              <el-button type="text">{{ tl('testsql') }}</el-button>
+              <el-button type="text">{{ tl('useSQL') }}</el-button>
+            </div>
+          </div>
+          <div class="input-container">
+            <el-input type="textarea" v-model="item.sql" :rows="10" />
+            <el-icon class="icon-copy"><copy-document /></el-icon>
+          </div>
+        </section>
+
+        <section>
+          <div class="sub-block-hd">
+            <p class="hd-title">{{ tl('usageScenarios') }}</p>
+          </div>
+          <p class="usage-desc">{{ item.scene }}</p>
+        </section>
+
+        <section>
+          <div class="sub-block-hd">
+            <p class="hd-title">{{ tl('exampleOfInput') }}</p>
+          </div>
+          <div class="input-container">
+            <el-input type="textarea" v-model="item.input" :rows="10" />
+            <el-icon class="icon-copy"><copy-document /></el-icon>
+          </div>
+        </section>
+
+        <section>
+          <div class="sub-block-hd">
+            <p class="hd-title">{{ tl('processedResults') }}</p>
+          </div>
+          <div class="input-container">
+            <el-input type="textarea" v-model="item.outputs" :rows="10" />
+            <el-icon class="icon-copy"><copy-document /></el-icon>
+          </div>
+        </section>
+      </el-collapse-item>
+    </el-collapse>
+  </el-drawer>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  name: 'SQLTemplateDrawer',
+})
+</script>
+
+<script setup lang="ts">
+import { ref, defineProps, computed, defineEmits, WritableComputedRef } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { CopyDocument } from '@element-plus/icons-vue'
+
+const { t } = useI18n()
+const tl = (key: string, moduleName = 'RuleEngine') => t(`${moduleName}.${key}`)
+
+const templateList = ref([
+  {
+    title: 'Process messages reported by devices',
+    scene: 'scene scene scene scene',
+    sql: 'SELECT',
+    input: '{"msg":100}',
+    outputs: '{"msg":100}',
```

Comment:
It is suggested that the data can be formatted using JSON stringify, with the following results:

```js
JSON.stringify({ msg: 100 }, null, 2)
```

```json
{
  "msg": 100
}
```